docs/03_tutorials/20_tic-tac-toe-game-smart-contract-single node.md

```diff
@@ -0,0 +1,761 @@
+    struct [[eosio::table]] game
+    {
+        static constexpr uint16_t boardWidth = 3;
+        static constexpr uint16_t boardHeight = boardWidth;
+        
+        game() : board(boardWidth * boardHeight, 0){}
+
+        name challenger, host, turn; // = account name of host, challenger and turn to store whose turn it is.
+        name winner = none; // = none/ draw/ name of host/ name of challenger
+
+        std::vector<uint8_t> board;
+
+        // Initialize the board with empty cell
+        void initializeBoard(){
+            board.assign(boardWidth * boardHeight, 0);
+        }
+
+        // Reset game
+        void resetGame(){
+            initializeBoard();
+            turn = host;
+            winner = "none"_n;
+        }
+
+        // 13. primary key accessor
+		auto primary_key() const { return challenger.value; }
+
+        //14. EOSLIB_SERIALIZE macro defining how the abi serializes / deserializes  
+		EOSLIB_SERIALIZE( game, (challenger)(host)(turn)(winner)(board))
+    };
+```
+
+15. Define the games type which uses the game data structure with the multi-index table template. Click on this link for more information about the multi-index table template used to define multi index table types [function multi_index.](https://developers.eos.io/manuals/eosio.cdt/v1.7/group__multiindex/#function-multi_index) Set the name to "games" and use the [eosio::name](https://developers.eos.io/manuals/eosio.cdt/v1.5/structeosio_1_1name) operator [_n](https://developers.eos.io/manuals/eosio.cdt/v1.5/name_8hpp#function-operator_n) to construct an [eosio::name](https://developers.eos.io/manuals/eosio.cdt/v1.5/structeosio_1_1name) with a string. The value is stored as a uint64_t. 
+    Add this code to the .hpp file inside the public section of the class, after the declaration of the game structure:
+
+```c++
+    // 15. Define the game data structure using the multi-index table template.
+    typedef eosio::multi_index<"games"_n, game> games;
+```
+
+16. Declare class methods and use the [[eosio::action]] attribute to let the compiler know this is a smart contract action. Click on this link for more information on [generator attributes.](https://developers.eos.io/manuals/eosio.cdt/v1.6/guides/generator-attributes) 
+    Add this code to the .hpp file inside the public section of the class:
+
+```c++
+    [[eosio::action]]
+    void create(const name &challenger, name &host);
+    
+    [[eosio::action]]
+    void restart(const name &challenger, const name &host, const name &by);
+
+    [[eosio::action]]
+    void close(const name &challenger, const name &host);
+    
+    [[eosio::action]]   
+    void move(const name &challenger, const name &host, const name &by, const uint16_t &row, const uint16_t &column);
+```
+
+17. The Move action uses the following private supporting methods to determine if a move is valid. They also check for a winning move:
+    * IsEmptyCell()
+    * IsValidMove()
+    * GetWinner()
```

Comment:
Use single codeblocks for these also. The trailing parentheses look weird otherwise.